---
title: "Untitled"
author: "Justine Guégan"
date: "26 septembre 2016"
output: pdf_document
---



L'objectif de ce TP est d'analyser un jeu de données 'RNASeq'. Il nécessite le chargement de plusieurs packages qui seront utilisés à différentes étapes du TP : 

- package DESeq2 : 

```{r echo=TRUE, eval = FALSE}
source("https://bioconductor.org/biocLite.R")
biocLite("DESeq2")
```

- package ggplot2

```{r echo=TRUE, eval = FALSE}
install.packages("ggplot2")
```

- package pheatmap

```{r echo=TRUE, eval = FALSE}
install.packages("pheatmap")
```

- package RColorBrewser

```{r echo=TRUE, eval = FALSE}
install.packages("RColorBrewser")
```



```{r load data}
library(DESeq2)
library(ggplot2)
library(pheatmap)
library(RColorBrewer)
comptage = read.delim("data/counts.txt",sep="\t", header=T,row.names=1)
countData = comptage[,-c(1:5)]
design = read.delim("data/annot_sample.csv",sep="\t", header=T,row.names=1)
colnames(countData) = rownames(design)
```


```{r create dds}
dds <- DESeqDataSetFromMatrix(countData = countData,
    colData = design,
    design = ~ condition)
```

How many genes have only 0 or 1 reads in all samples ? --> Remove those genes
```{r lowCounts}
dds <- dds[ rowSums(counts(dds)) > 1, ]

```

```{r analysis}
dds$condition <- relevel(dds$condition, ref="NBS")
dds <- DESeq(dds)

```

Est- ce que toutes les librairires font la même taille ? Repésentation barplot des comptages, boxplot avant/après normalisation

```{r normalisation}
barplot(colSums(counts(dds)))
sizeFactors(dds) # divide each column by his size factor --> normalized counts 

par(mfrow=c(1,2))
boxplot(log2(counts(dds) +1 ), las=2, col=as.numeric(design$condition)+1)
boxplot(log2(counts(dds, normalized=TRUE) +1 ),las=2, col=as.numeric(design$condition)+1)

```

PCA :

```{r pca}
rld = rlog(dds, blind=FALSE)
plotPCA(rld, intgroup=c("condition"),ntop=nrow(rld))

acp = prcomp(t(assay(rld)),center=TRUE)
plot(acp$x[,1], acp$x[,2], col=as.numeric(design$condition), pch=16)
text(acp$x[,1]+3, acp$x[,2]+3, col=as.numeric(design$condition),labels = colnames(rld))

```

Heatmap sample sample

```{r Heatmap}
sampledist = dist(t(assay(rld)))
sampledistMatrix =as.matrix(sampledist)
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampledistMatrix,col=colors)
```

Analyse Diff

```{r analyse diff}
resHER2 <- results(dds, contrast=c("condition","HER2","NBS"),alpha = 0.001)
resHER2Ordered <- resHER2[order(resHER2$padj),]
summary(resHER2)
```


```{r volcano}
up = which(resHER2$pvalue <= 0.001 & resHER2$log2FoldChange >= 1)
down = which(resHER2$pvalue <= 0.001 & resHER2$log2FoldChange <= -1)

plot(resHER2$log2FoldChange, -log10(resHER2$pvalue), type="n", xlab="log2Fold-Change", ylab="-log10(pvalue)", main="HER2 vs NBS")
points(resHER2$log2FoldChange[up], -log10(resHER2$pvalue[up]),col="red",pch=16)
points(resHER2$log2FoldChange[down], -log10(resHER2$pvalue[down]),col="green",pch=16)
points(resHER2$log2FoldChange[-c(up,down)], -log10(resHER2$pvalue[-c(up,down)]), pch=16)

abline(v=c(-1,1), lty=2)
abline(h=-log10(0.001), lty=2)
```


Dessiner le profil du gène clef dans les cancers HER2 (ERBB2) comparé aux autres types de cancer et aux éch NBS

```{r ERBB2}
plotCounts(dds, gene="ERBB2", intgroup="condition")
```

